### PR TITLE
[DI][FrameworkBundle] Use container.hidden tag to hide services from debug:container

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #26630
| License       | MIT
| Doc PR        | -

Now that services are private by default, the `debug:container` should display them by default.
In fact, the public/private concept should not trigger a difference in visibility for this command.
Instead, I propose to handle a new `container.hidden` tag, for the purpose of tagging services as hidden (~internal).

I now need to apply the tag to more services created internally.

Any comments at this stage?

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php [hunk 1]
```diff
@@ -103,14 +99,18 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        if ($input->getOption('show-private')) {
+            @trigger_error('The "--show-private" option is a no-op and is deprecated since Symfony 4.1.', E_USER_DEPRECATED);
```
Comment (weaverryan) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php]: > option no longer has any effect and is deprecated ...

**Review by weaverryan — COMMENTED** [review]
Obviously, you know I like this idea. Particularly, it's important to now show ALL services in `debug:container`. The ability to then hide a few of the *most* internal, is great.

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/Descriptor.php [hunk 2]
```diff
@@ -230,27 +231,34 @@ protected function resolveServiceDefinition(ContainerBuilder $builder, $serviceI
             return $builder->getAlias($serviceId);
         }
 
+        if ('service_container' == $serviceId) {
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/Descriptor.php]: should use a strict comparison

**Review by stof — COMMENTED** [review]
The new tag must be added in the whitelist of the UnusedTags pass, to avoid complaining about this tag not being during the container building.

**Comment by nicolas-grekas** [comment]
Replaced by #26921

**Pull request closed without merging**